rest-api-json-bulk/src/main/kotlin/org/modelix/sample/restapijsonbulk/BulkApi.kt

```diff
@@ -34,30 +36,32 @@ object RouteHelper {
 fun Route.BulkApi(loadRoots: suspend () -> List<INode>, resolve: suspend (INodeReference) -> INode?) {
     get<Paths.getLectures> {
         val roots = loadRoots()
+
         val allLectures = roots.flatMap {
-            (it.allChildren.map { MPSLanguageRegistry.getInstance<BaseConcept>(it) })
-        }.filterIsInstance<Courses>().flatMap { it.children.lectures }
+            (it.allChildren.map { it.typed() })
+        }.filterIsInstance<N_Courses>().flatMap { it.lectures }
 
-        val roomList = LectureList(lectures = allLectures.map {
-            Lecture(name = it.properties.name!!,
-                    description = it.properties.description!!,
-                    lectureRef = RouteHelper.urlEncode(INodeReferenceSerializer.serialize(it.iNode.reference)),
-                    room = RouteHelper.urlEncode(INodeReferenceSerializer.serialize(it.references.room.iNode.reference)),
-                    maxParticipants = it.properties.maxParticipants!!)
+        val roomList = LectureList(lectures = allLectures.map { lectureInstance ->
```

Comment:
why not just `lecture`?

Reply:
I think the SPA specifically expects a LecutreList in return? Don't know how an alternative could look like?

Reply:
I was referring to the variable name:

```suggestion
        val roomList = LectureList(lectures = allLectures.map { lecture ->
```

And so on